rpc_util_test.go

```diff
@@ -290,3 +293,115 @@ func BenchmarkGZIPCompressor512KiB(b *testing.B) {
 func BenchmarkGZIPCompressor1MiB(b *testing.B) {
 	bmCompressor(b, 1024*1024, NewGZIPCompressor())
 }
+
+type MockCompressor struct {
+	DecompressedData []byte
+	ErrDecompress    error
+	Size             int
+	CustomReader     io.Reader
+}
+
+func (m *MockCompressor) Compress(w io.Writer) (io.WriteCloser, error) {
+	return nil, nil
+}
+
+func (m *MockCompressor) Decompress(r io.Reader) (io.Reader, error) {
+	if m.ErrDecompress != nil {
+		return nil, m.ErrDecompress
+	}
+	if m.CustomReader != nil {
+		return m.CustomReader, nil
+	}
+	return bytes.NewReader(m.DecompressedData), nil
+}
+
+func (m *MockCompressor) DecompressedSize(compressedBytes mem.BufferSlice) int {
+	return m.Size
+}
+
+func (m *MockCompressor) Name() string {
+	return "mockCompressor"
+}
+
+type ErrorReader struct{}
+
+func (e *ErrorReader) Read(p []byte) (n int, err error) {
+	return 0, errors.New("simulated io.Copy read error")
+}
+
+func TestDecompress(t *testing.T) {
+	tests := []struct {
+		name                  string
+		compressor            encoding.Compressor
+		input                 mem.BufferSlice
+		maxReceiveMessageSize int
+		wantedOutput          mem.BufferSlice
+		wantedSize            int
+		wantedError           error
+	}{
+		{
+			name: "Successful decompression",
+			compressor: &MockCompressor{
+				DecompressedData: []byte("decompressed data"),
+				Size:             17,
+			},
+			input:                 mem.BufferSlice{}, // You can fill in the actual input here
+			maxReceiveMessageSize: 100,
+			wantedOutput: func() mem.BufferSlice {
+				decompressed := []byte("decompressed data")
+				return mem.BufferSlice{mem.NewBuffer(&decompressed, nil)} // Taking the address of the slice variable
+			}(),
+			wantedSize:  1,
+			wantedError: nil,
+		},
+		{
+			name: "Error during decompression",
+			compressor: &MockCompressor{
+				ErrDecompress: errors.New("decompression error"),
+			},
+			input:                 mem.BufferSlice{},
+			maxReceiveMessageSize: 100,
+			wantedOutput:          nil,
+			wantedSize:            0,
+			wantedError:           errors.New("decompression error"),
+		},
+		{
+			name: "Buffer overflow",
```

Comment:
We need one more test case where maxReceiveSize is math.MaxInt64 but the actual size of data is way low. This is the fix we are trying to fix so you need test case for that

Reply:
added  math.Maxint64  test condition